### PR TITLE
Sortable products in collections

### LINKED ISSUE #3900: Ability to order products in a collection
### What I'm trying to achieve
As a store admin, I'd like to have more control over how the products are displayed in the storefront, so I can put bestsellers or products that I want to get rid off at the top of the page

### LINKED ISSUE #4049: Ability to order products in a category
As a store admin, I'd like to have more control over how the products are displayed in the storefront, so I can put bestsellers or products that I want to get rid off at the top of the page.

Categories have a tree-like structure, not sure how we want to solve this, but for example, ordering could happen only at the top level.

### PR DESCRIPTION
**Pull request by NyanKiyoshi**
- **Definitions**
  - [x] ~~Categories contains sortable products -> `category_sort -> qs = category.products`~~
  - [x] Collections contains sortable products -> Many2Many - through: CollectionProducts -> product + collection
  - [x] References are properly deleted/cleaned and handled, no traces are left when a change is made to relation
  - [x] ~~Can sort by category custom sort~~
  - [x] Can sort by collection custom sort
  - [x] ~~Can retrieve the sort position of a product inside a category (implicit, only the relative position)~~
  - [x] Can retrieve the sort position of a product inside a collection (implicit, only the relative position)
  - [ ] ~~Can be able to sort products within a collection/category through filters~~ edit: #4143
  - [x] Migrations are done and migrates old data, if needed

- **GraphQL Mutations**
  - [x] `ProductCategoryReorderBulk (bulk)`
  - [x] `ProductCollectionReorderBulk (bulk)`

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [x] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### Magic Words
I summon with my magic wand... closes #3900.

### CODE REVIEW

File: saleor/graphql/product/mutations/products.py [hunk 1]
```diff
@@ -191,6 +204,75 @@ class Meta:
...
+            product_id = from_global_id_strict_type(
+                info, move_info.product_id, only_type=Product, field="moves"
+            )
+
+            try:
+                node = models.CollectionProduct.objects.get(
+                    product_id=product_id, collection_id=collection_id
+                )
+            except models.CollectionProduct.DoesNotExist:
+                raise ValidationError(
```
Comment (NyanKiyoshi) [inline saleor/graphql/product/mutations/products.py]: Need to test this.

File: saleor/product/models.py [hunk 2]
```diff
@@ -192,6 +192,9 @@ def get_price_range(self, discounts=None, taxes=None):
         price = apply_tax_to_price(taxes, tax_rate, price)
         return TaxedMoneyRange(start=price, stop=price)
 
+    def get_ordering_queryset(self):
```
Comment (NyanKiyoshi) [inline saleor/product/models.py]: Dead code.

File: saleor/graphql/utils.py [hunk 3]
```diff
@@ -78,7 +78,10 @@ def get_nodes(ids, graphene_type=None):
     if nodes_type and not graphene_type:
         graphene_type = _resolve_graphene_type(nodes_type)
 
-    nodes = list(graphene_type._meta.model.objects.filter(pk__in=pks))
+    if qs is None:
```
Comment (NyanKiyoshi) [inline saleor/graphql/utils.py]: dead code.

File: saleor/menu/models.py [hunk 4]
```diff
@@ -27,7 +27,11 @@ def __str__(self):
         return self.name
 
 
-class MenuItem(MPTTModel, SortableModel):
+class _MenuItemMeta(type(MPTTModel), type(SortableModel)):
```
Comment (NyanKiyoshi) [inline saleor/menu/models.py]: dead code, no longer needed.

**Comment by maarcingebala** [comment]
Looks great @NyanKiyoshi!

**Pull request merged**